builtin/providers/ibmcloud/provider.go

```diff
@@ -0,0 +1,81 @@
+package ibmcloud
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// Provider returns a terraform.ResourceProvider.
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"ibmid": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The IBM ID.",
+				DefaultFunc: schema.EnvDefaultFunc("IBMID", ""),
+			},
+			"ibmid_password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The password for the IBM ID.",
+				DefaultFunc: schema.EnvDefaultFunc("IBMID_PASSWORD", ""),
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The Bluemix Region (for example 'ng').",
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"BM_REGION", "BLUEMIX_REGION"}, "ng"),
+			},
+			"softlayer_timeout": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The timeout (in seconds) to set for any SoftLayer API calls made.",
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"SL_TIMEOUT", "SOFTLAYER_TIMEOUT"}, 60),
+			},
+			"softlayer_account_number": {
```

Comment:
The current provider will cut across the IBM Cloud portfolio and it just helps to point out that this particular account number is related to SoftLayer.